File: src/main/java/com/severett/chargerapp/service/SessionService.java
package com.severett.chargerapp.service;

import com.severett.chargerapp.model.ChargingSession;
import com.severett.chargerapp.model.Summary;

import java.time.Instant;
import java.util.List;

public interface SessionService {

    ChargingSession createSession(String stationId);

    ChargingSession stopSession(String id) throws IllegalArgumentException;

    List<ChargingSession> getSessions();

    Summary getSummary();

}

File: src/main/kotlin/com/severett/chargerapp/service/SessionService.kt
package com.severett.chargerapp.service

import com.severett.chargerapp.model.ChargingSession
import com.severett.chargerapp.model.Summary
import java.time.Instant
import java.util.*

interface SessionService {

    fun createSession(id: UUID, stationId: String?, timestamp: Instant): ChargingSession

    fun stopSession(id: UUID, timestamp: Instant): ChargingSession?

    fun getSessions(): List<ChargingSession>

    fun getSummary(fromTimestamp: Instant, toTimestamp: Instant): Summary

}
